optimum/habana/transformers/generation/utils.py

```diff
@@ -1097,12 +1100,21 @@ def greedy_search(
             model_inputs = self.prepare_inputs_for_generation(input_ids, **model_kwargs)
 
             # forward pass to get next token
-            outputs = self(
-                **model_inputs,
-                return_dict=True,
-                output_attentions=output_attentions,
-                output_hidden_states=output_hidden_states,
-            )
+            if enable_trim_logits:
+                outputs = self(
+                    **model_inputs,
+                    return_dict=True,
+                    output_attentions=output_attentions,
+                    output_hidden_states=output_hidden_states,
+                    trim_logits=model_kwargs["trim_logits"],
+                )
+            else:
+                outputs = self(
+                    **model_inputs,
+                    return_dict=True,
+                    output_attentions=output_attentions,
+                    output_hidden_states=output_hidden_states,
+                )
```

Comment:
Once `trim_logits` is part of the model inputs, we should not need to do that.

Reply:
Got it, fixed accordingly, thank you!